### PR TITLE
add Range to NewIterator

### PR DESCRIPTION
**Pull request by klim0v**
I see a slowdown when using iterations in the https://github.com/cosmos/iavl/pull/324#issuecomment-723961546 library in the case where the given key cannot be found. I guess this is because the `newGoLevelDBIterator` function uses `Seek()`, which has to go through the whole tree to make sure the key is missing.

### CODE REVIEW

**Comment by klim0v** [comment]
> Error: requested golangci-lint version 'v1.26' isn't supported: we support only v1.28.3 and later versions

Can you fix the lint/golangci config?

**Comment by marbar3778** [comment]
> Can you fix the lint/golangci config?

Could you make this PR against master, this error won't arise there, after that it can be back ported?

**Comment by melekes** [comment]
wow, what a great find!

File: goleveldb.go [hunk 1]
```diff
@@ -175,7 +176,7 @@ func (db *GoLevelDB) Iterator(start, end []byte) (Iterator, error) {
 	if (start != nil && len(start) == 0) || (end != nil && len(end) == 0) {
 		return nil, errKeyEmpty
 	}
-	itr := db.db.NewIterator(nil, nil)
+  itr := db.db.NewIterator(&util.Range{Start: start, Limit: end}, nil)
```
Comment (erikgrinaker) [inline goleveldb.go:179]:
```suggestion
	itr := db.db.NewIterator(&util.Range{Start: start, Limit: end}, nil)
```

**Comment by erikgrinaker** [comment]
Can you add an entry for this in `CHANGELOG_PENDING.md` as well please?

File: goleveldb.go (hunk 1 above)
Comment (klim0v) [reply goleveldb.go]: resolve

File: goleveldb.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package db
 
 import (
 	"fmt"
+	"github.com/syndtr/goleveldb/leveldb/util"
 	"path/filepath"
 
 	"github.com/syndtr/goleveldb/leveldb"
```
Comment (erikgrinaker) [inline goleveldb.go:8]:
```suggestion
	"path/filepath"

	"github.com/syndtr/goleveldb/leveldb"
	"github.com/syndtr/goleveldb/leveldb/util"
```

**Comment by klim0v** [comment]
> Can you add an entry for this in `CHANGELOG.md` as well please?

It's like that?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- [\#134](https://github.com/tendermint/tm-db/pull/134) Add the `util.Range` to `NewIterator` (@klim0v)
```
Comment (erikgrinaker) [inline CHANGELOG.md:5]:
```suggestion
- [\#134](https://github.com/tendermint/tm-db/pull/134) Improve GoLevelDB iterator performance by bounding underlying iterator range (@klim0v)
```

**Review by erikgrinaker — APPROVED** [review]

**Comment by erikgrinaker** [comment]
Thanks! I'll run some tests with IAVL and Tendermint using this change, and if it looks good I'll backport to 0.5.x and cut new releases - won't be able to do this until tomorrow though, since I need to finish up some other work tonight.

**Comment by klim0v** [comment]
Thank you, I was happy to contribute to the common cause!

**Pull request merged**